File: Day 5/Part 2.js
var otot = java.util.regex.Pattern.compile("(.)(.).*\\1\\2");
var oto  = java.util.regex.Pattern.compile("(.).\\1");

function one_two_one_two(str) {
    return otot.matcher(str).find();
}

function one_two_one(str) {
    return oto.matcher(str).find();
}

function is_nice(str) {
    return one_two_one_two(str) && one_two_one(str);
}

function find_nice() {
    var line;
    var count = 0;
    var file = new java.io.File("input");
    var reader = new java.io.BufferedReader(new java.io.FileReader(file));
    while ((line = reader.readLine()) != null)
        if (is_nice(line))
            count++;
    print("There are " + count + " nice strings.");
}

find_nice()

File: Day 8/Part 2.kt
package xyz.nickr

fun main(args : Array<String>) {
    val file = java.io.File("input")
    val reader = file.bufferedReader()
    var line = reader.readLine()
    var clens = 0
    var alens = 0
    while (line != null) {
     
        clens += line.length
        alens += ('"' + line.replace("\\", "\\\\").replace("\"", "\\\"") + '"').length

        line = reader.readLine()

    }
    println(alens - clens)
}